src/crab/type_encoding.hpp

```
@@ -51,6 +57,89 @@ std::string typeset_to_string(const std::vector<TypeEncoding>& items);
+/// by TypeEncoding).
+class TypeSet {
+    SmallBitsetDomain<uint8_t> dom_;
+
+  public:
+    constexpr TypeSet() = default;
+    constexpr explicit TypeSet(uint8_t bits) : dom_{bits} {}
+
+    /// The empty set.
+    static constexpr TypeSet empty() { return TypeSet{0}; }
+    /// The full set (all 8 types).
+    static constexpr TypeSet all() { return TypeSet{0xFF}; }
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)